### PR TITLE
[ErrorRenderer] Refactor and delegate serializing responsibility to the Serializer component

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | TODO

By default, the ErrorRenderer component will host only the HTML+JS error page, delegating the serialization of other formats to the Serializer component (if installed & serializer config is enabled). This will simplify the usage and integration with other third-party packages like API-Platform and FOSRestBundle.

**Main changes:**
 * Added new `ProblemNormalizer` to the Serializer component ([RFC 7807](https://tools.ietf.org/html/rfc7807) compliant), 
 * Removed `<Format>ErrorRenderer` and registration mechanism in favor of Serializer's normalizers (see `UPGRADE-4.4.md` changes),
 * Removed `debug:error-renderer` command, it's no longer needed, as well as other classes.
 * The response structure was adjusted according to the one implemented in API-Platform https://github.com/api-platform/core/blob/master/src/Problem/Serializer/ErrorNormalizer.php

Related comment:
 * https://github.com/symfony/symfony/pull/33650#issuecomment-534441889
 * also discussed with @nicolas-grekas and @dunglas on Slack.

wdyt?

### CODE REVIEW

File: UPGRADE-4.4.md [hunk 1]
```diff
@@ -18,7 +18,6 @@ Console
 Debug
 -----
 
- * Deprecated the `Debug` class, use the one from the `ErrorRenderer` component instead
```
Comment (yceruto) [inline UPGRADE-4.4.md]: [duplicated] the whole component is deprecated (mentioned two lines bellow)

File: UPGRADE-5.0.md [hunk 2]
```diff
@@ -57,7 +57,6 @@ Console
 Debug
 -----
 
- * Removed the `Debug` class, use the one from the `ErrorRenderer` component instead
```
Comment (yceruto) [inline UPGRADE-5.0.md]: same here

File: src/Symfony/Component/ErrorRenderer/Exception/FlattenException.php [hunk 3]
```diff
@@ -26,14 +26,14 @@
  */
 class FlattenException extends LegacyFlattenException
 {
-    private $title;
```
Comment (yceruto) [inline src/Symfony/Component/ErrorRenderer/Exception/FlattenException.php]: minor refactor for better name (it was introduced in 4.4)

**Comment by yceruto** [comment]
Ready for review!

(Travis dep=high failures is normal)

File: src/Symfony/Component/Serializer/Normalizer/ProblemNormalizer.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        $this->defaultContext = $defaultContext + $this->defaultContext;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($exception, $format = null, array $context = [])
+    {
+        if (!$exception instanceof FlattenException) {
+            $exception = FlattenException::createFromThrowable($exception);
```
Comment (fabpot) [inline src/Symfony/Component/Serializer/Normalizer/ProblemNormalizer.php]: What if ErrorRenderer is not installed?

**Review by fabpot — CHANGES_REQUESTED** [review]
Great work!

File: UPGRADE-4.4.md [hunk 5]
```diff
@@ -307,45 +306,35 @@ TwigBundle
...
-                   'code' => $exception->getStatusCode(),
-                   'message' => $exception->getMessage(),
-               ],
-           ]);
+           return 'json' === $format && $data instanceof FlattenException;
        }
    }
    ```
 
-  Configure your rendering service tagging it with `error_renderer.renderer`.
```
Comment (chalasr) [inline UPGRADE-4.4.md]: Should be removed from:
https://github.com/symfony/symfony/blob/d5ba53547e25036a3473fc33d918c485ff85018b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/UnusedTagsPass.php#L35

**Comment by yceruto** [comment]
closing in favor of https://github.com/symfony/symfony/pull/34312

**Pull request closed without merging**